### PR TITLE
Java Blob fixes

### PR DESCRIPTION
**Pull request by kirkeaton**
1. Fixed BlobStore to write all incoming data
2. Removed HttpServletRequestBlob.java
3. Removed InputStreamBlob
4. Modified HttpBlobService to implement WebAgent
5. Modified upload logic to store a temporary file before putting to BlobStore
6. Added BlobBlob class
7. Modified RestBlobService to return a BlobBlob
8. Updated address in RestBlobService to include /service prefix

### CODE REVIEW

File: src/foam/blob/BlobBlob.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.blob;
+
+import java.nio.ByteBuffer;
+
+public class BlobBlob
```
Comment (kgrgreer) [inline src/foam/blob/BlobBlob.java]: Why is this called a BlobBlob? Wouldn't ByteArrayBlob be a better name?

File: src/foam/nanos/blob/HttpBlobService.java [hunk 2]
```diff
@@ -7,125 +7,127 @@
...
-
-  @Override
-  public String urlFor(Blob blob) {
-    return this.urlFor_(getX(), blob);
+  public void execute(X x) {
+    HttpServletRequest  req  = (HttpServletRequest)  x.get(HttpServletRequest.class);
+    HttpServletResponse resp = (HttpServletResponse) x.get(HttpServletResponse.class);
+
+    try {
+      if ("GET".equals(req.getMethod())) {
```
Comment (kgrgreer) [inline src/foam/nanos/blob/HttpBlobService.java]: Add spaces to conform to FOAM's style guidelines.

File: src/foam/blob/BlobBlob.java (hunk 1 above)
Comment (kirkeaton) [reply src/foam/blob/BlobBlob.java]: @kgrgreer BlobBlob is what the Javascript equivalent is called

File: src/foam/nanos/blob/HttpBlobService.java [hunk 3]
```diff
@@ -7,125 +7,127 @@
...
+        download(req, resp);
+      } else if ("PUT".equals(req.getMethod())) {
```
Comment (kgrgreer) [inline src/foam/nanos/blob/HttpBlobService.java]: spacing

File: src/foam/nanos/blob/HttpBlobService.java [hunk 4]
```diff
@@ -7,125 +7,127 @@
...
-  public foam.core.X getX() {
-    return x_;
-  }
+      Blob blob = getDelegate().find(id);
 
-  @Override
-  public void setX(foam.core.X x) {
-    x_ = x;
-  }
+      if (blob == null) {
```
Comment (kgrgreer) [inline src/foam/nanos/blob/HttpBlobService.java]: spacing

File: src/foam/nanos/blob/HttpBlobService.java [hunk 5]
```diff
@@ -7,125 +7,127 @@
...
-  @Override
-  public void start() {
-    store_ = (foam.blob.BlobService)getX().get("blobStore");
+      long chunk = 0;
+      long size = blob.getSize();
+      long chunks = (long) Math.ceil((double) size / (double) BUFFER_SIZE);
+      Buffer buffer = new Buffer(BUFFER_SIZE, ByteBuffer.allocate(BUFFER_SIZE));
+
+      resp.setStatus(resp.SC_OK);
+      if (blob instanceof FileBlob) {
```
Comment (kgrgreer) [inline src/foam/nanos/blob/HttpBlobService.java]: spacing, here and elsewhere

Comment (kirkeaton) [reply src/foam/nanos/blob/HttpBlobService.java]: Fixed the spacing issues in this class

File: src/foam/blob/BlobBlob.java (hunk 1 above)
Comment (kirkeaton) [reply src/foam/blob/BlobBlob.java]: @kgrgreer Renamed to ByteArrayBlob

**Pull request merged**